### PR TITLE
tests: add tests for slow and error responses from user and statuspage

### PR DESCRIPTION
**Pull request by ciyer**
Work on #2283

### CODE REVIEW

**Comment by ciyer** [comment]
I added the tests suggested in [#2283](https://github.com/SwissDataScienceCenter/renku-ui/issues/2283), but I still do not see the behavior described in that issue. Though I believe all those that have reported it, from looking at the code, it isn't clear how this situation could happen in the way described

I think a more likely scenario that could cause this would be an error response from the `user` endpoint, followed by a UI refresh, followed by a valid response from `user`.

File: tests/cypress/e2e/home.spec.ts [hunk 1]
```diff
@@ -37,6 +37,28 @@ describe("display the home page", () => {
   });
 });
 
+describe("display the home page even when APIs return strange responses", () => {
+  beforeEach(() => {
+    new Fixtures(cy)
+      .config()
+      .versions()
+      .statuspageDown()
+      .userNone("getUser", 5000);
```
Comment (lorenzo-cavazzi) [inline tests/cypress/e2e/home.spec.ts:46]: This can probably be much lower than 5000, I'd say 1000

**Review by lorenzo-cavazzi — COMMENTED** [review]
I suggest having a single place where we verify the maintenance messages work as expected.
Perhaps we should move the maintenance-related tests (that includes the `"display the maintenance page"` section, to be updated or merged with the new tests) to the maintenace.spec.ts file and let the home specs check only home page-specific components.

**Comment by ciyer** [comment]
> I suggest having a single place where we verify the maintenance messages work as expected. Perhaps we should move the maintenance-related tests (that includes the `"display the maintenance page"` section, to be updated or merged with the new tests) to the maintenace.spec.ts file and let the home specs check only home page-specific components.

I moved the `display maintenance page` test to the `mainenance.spec.ts` file, but I kept the new test in the home page, since it tests that the home page, and not the mainenance page, is shown. I reduced the delay to 1s, as suggested.

File: tests/cypress/e2e/maintenance.spec.ts [hunk 2]
```diff
@@ -45,6 +45,35 @@ describe("display the maintenance page", () => {
   });
 });
 
+describe("display the maintenance page when there is no user response", () => {
+  beforeEach(() => {
+    new Fixtures(cy).config().versions();
+  });
+
+  it("displays an error when trying to get status page information", () => {
+    // ! we plan to change this behavior and ignore statuspage info when unavailable #2283
```
Comment (lorenzo-cavazzi) [inline tests/cypress/e2e/maintenance.spec.ts:54]:
```suggestion
```
We can remove this line

**Review by lorenzo-cavazzi — CHANGES_REQUESTED** [review]
Lgtm!

Before merging, let's remove the comment about the #2283 issue since it's closed by this PR.

**Comment by lorenzo-cavazzi** [comment]
The test `display the home page even when APIs return strange responses` seems to fail consistently -- I tried it locally as well

**Comment by ciyer** [comment]
> The test `display the home page even when APIs return strange responses` seems to fail consistently -- I tried it locally as well

I see that now -- it was a problem introduced as a side effect of merging changes from master. Everything should work now.

**Review by lorenzo-cavazzi — APPROVED** [review]
Lgtm!

**Pull request merged**